### DISCUSSION TITLE
AS4ClientUserMessage how to support outboundUserName/Password.

### DISCUSSION THREAD

**Discussion by TMain-Tomi** [phax/phase4#135] [Q&A]
Dear @phax ,

I used phase4-spring-boot-demo to start AS4 server and client.
I used **AS4ClientUserMessage** and **AS4ClientSentMessage** to push a msg to partner. and my code as follow, it works normal.
```
as4ClientUserMessage.addAttachment(file1, CMimeType.TEXT_PLAIN, null)
as4ClientUserMessage = BaseDataPhase4.setKeyStore(as4ClientUserMessage, as4Partner)
as4ClientUserMessage.signingParams().setAlgorithmSign(ECryptoAlgorithmSign.getFromIDOrDefault(as4Partner.sign, ECryptoAlgorithmSign.SIGN_ALGORITHM_DEFAULT)).setAlgorithmSignDigest(ECryptoAlgorithmSignDigest.getFromIDOrDefault(digest, ECryptoAlgorithmSignDigest.SIGN_DIGEST_ALGORITHM_DEFAULT));                          
as4ClientUserMessage.cryptParams().setAlgorithmCrypt(ECryptoAlgorithmCrypt.getFromIDOrDefault(as4Partner.encrypt, ECryptoAlgorithmCrypt.ENCRPYTION_ALGORITHM_DEFAULT))
ResponseHandlerMicroDom responseHandlerMicroDom = new ResponseHandlerMicroDom()
IAS4OutgoingDumper dumper = AS4DumpManager.getOutgoingDumper()
AS4ClientSentMessage as4ClientSentMessage = as4ClientUserMessage.sendMessageWithRetries(as4Partner.protocolAddress, responseHandlerMicroDom, null, dumper, null);
```
**But sometimes customers need we take outboundUserName/Password to tighten security. does as4ClientUserMessage can take outboundUserName/Password?
Similarly, if I were a server, if customers take inboundUserName/Password in msg, how do I parse it?**
May you give me some advices? thanks.

Best Regards and Thanks.
Tomi Lin

**Comment by phax** [comment]
Hi Tomi Lin, Username and Password are currently not supported. See also #42 
A Pull Request is always welcome :)

**Reply by TMain-Tomi** [reply]
Hi phax ,

Acknowledged, thanks for your reply.

Best Regards and Thanks.
Tomi Lin
